### ISSUE TITLE
When balls are removed from the board, should we still count them towards total N? 

### ISSUE THREAD

**Issue by phet-steele** [phetsims/plinko-probability#85]
Very similar to #27, and may be a consequence of how it was handled.

1. Lab screen, select single ball play mode with "Ball" hopper mode. From here it may be useful to have 5 rows and either 0 or 1 as the binary probability. Additionally, use the `?maxBalls` query with a low value (25) so there is less waiting time.
2. Play one ball at a time up to one below your set max (24). Read the entire next step before following it.
3. Press the play button one more time and quickly close the "Out of Balls" dialogue. Still quickly, change to a different hopper mode **while the ball is still traveling down the board**.

The ball will have been removed from the board, but not count towards the histogram. The sim still believes a histogram is at max and will keep the Play button disabled. I'd think that removed balls still need to count towards the quantity shown in the histograms in this case (but maybe all cases?).

![pp04](https://cloud.githubusercontent.com/assets/5863899/18563251/35f54c7c-7b44-11e6-86d0-a21ca26561f1.gif)

Seen on OSX 10.11.6 Safari. For phetsims/tasks/issues/696.
Name: ‪Plinko Probability‬
URL: http://www.colorado.edu/physics/phet/dev/html/plinko-probability/1.0.0-dev.16/plinko-probability_en.html?maxBalls=25
Version: 1.0.0-dev.16 2016-09-13 20:39:43 UTC
Features missing: touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/601.7.8 (KHTML, like Gecko) Version/9.1.3 Safari/601.7.8
Language: en-us
Window: 1920x1036
Pixel Ratio: 2/1
WebGL: WebGL 1.0 (2.1 ATI-1.42.15)
GLSL: WebGL GLSL ES 1.0 (1.20)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 32 uniform: 1024
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 16)
Max viewport: 16384x16384
OES_texture_float: true
Dependencies JSON: {"assert":{"sha":"7d27130a","branch":"master"},"axon":{"sha":"e0192608","branch":"master"},"babel":{"sha":"ca508dea","branch":"master"},"brand":{"sha":"f0b1f7da","branch":"master"},"chipper":{"sha":"9f0bb7ca","branch":"master"},"dot":{"sha":"39436598","branch":"master"},"joist":{"sha":"752d3a03","branch":"master"},"kite":{"sha":"73302899","branch":"master"},"phet-core":{"sha":"c48bf320","branch":"master"},"phetcommon":{"sha":"83ea84c8","branch":"master"},"plinko-probability":{"sha":"cbd8dd29","branch":"master"},"query-string-machine":{"sha":"20686a62","branch":"master"},"scenery":{"sha":"e78ee413","branch":"master"},"scenery-phet":{"sha":"f0fc9ae8","branch":"master"},"sherpa":{"sha":"bcc28cd6","branch":"master"},"sun":{"sha":"459de68b","branch":"master"},"tandem":{"sha":"4a8edbc9","branch":"master"},"vibe":{"sha":"b422db9c","branch":"master"}}

**Comment by pixelzoom** [comment]
Yes, I noticed this when testing the "Out of Balls" dialog.  The code that keeps track of the number of balls is certainly a bit suspect, and not at all obvious how/where to fix this.  That said... 

The max number of balls is 9999, so few people are likely to see the dialog. And if they do see the dialog, they won't know whether the max is 9998 or 9999 - they simply know that they're out of balls.

So my feeling is that we shouldn't bother trying to figure this one out. @ariel-phet and @amanda-phet, your opinion?

**Comment by ariel-phet** [comment]
I would be fine punting on this one, I don't think it is likely to cause anyone confusion.

I could see a teacher letting the sim run in the background, and saying "lets come back to this later" and let it get to max. So some people might get to the max...but the actual value of the max is not actually important. So it seems like this is an edge case we don't need to address at this time. 

Perhaps when we work on the number of rows we can revisit the code that keeps track of the number of balls? (so for V2)

**Comment by amanda-phet** [comment]
I would definitely expect a teacher to let the sim run in the background so that they can demonstrate the effect of a large N. However, the goal of the max number of balls is simply to avoid a bin with more than 9999 balls (due to the size of the text decreasing with each additional digit). As long as users can run the sim to the max number of balls without running into a real problem, I am ok with a few balls not being counted in N.

**Comment by amanda-phet** [comment]
(@ariel-phet what should I do with this issue?)

**Comment by ariel-phet** [comment]
@amanda-phet removing your assignment, we will leave this issue until we revisit the number of rows

**Comment by pixelzoom** [comment]
OK, thanks. Deferred for 1.0 release, to be addressed in a future release.

**Comment by pixelzoom** [comment]
@ariel-phet said:
> ... we will leave this issue until we revisit the number of rows

FYI... Number of rows was addressed (and closed) in https://github.com/phetsims/plinko-probability/issues/84, but this issue was apparently not consulted.
